logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -126,6 +146,42 @@ def config_init(params)
     @config = params
   end # def config_init
 
+  # Replace all environment variable references in 'value' param by environment variable value and return updated value
+  # Process following patterns : $VAR, ${VAR}, ${VAR:defaultValue}
+  def replace_env_placeholders(value)
+    if (value.is_a?(String))
+      while value =~ ENV_PLACEHOLDER_REGEX
```

Comment:
I agree that using named captures is cleaner !
And don't worry, you was not confusing at all, I have fully understood what you mean :)

So it's ok to me to do like that.